test/locking.js

```diff
@@ -44,196 +42,291 @@ contract('Staking app, Locking', ([owner, user1, user2]) => {
   })
 
   it('locks', async () => {
-    const lockId = await approveStakeAndLock(user1)
+    await approveStakeAndLock(user1)
 
     // check lock values
-    const [ _amount, _unlockedAt, _manager, _data ] = await staking.getLock(owner, lockId)
+    const [ _amount, _allowance ] = await staking.getLock(owner, user1)
     assert.equal(_amount, DEFAULT_LOCK_AMOUNT, "locked amount should match")
-    assert.equal(_unlockedAt.toString(), MAX_UINT64.toString(), "unlock time should match")
-    assert.equal(_manager, user1, "unlocker should match")
-    assert.equal(_data, zeroBytes, "lock data should match")
+    assert.equal(_allowance, DEFAULT_LOCK_AMOUNT, "locked allowance should match")
 
     assert.equal((await staking.unlockedBalanceOf(owner)).valueOf(), DEFAULT_STAKE_AMOUNT - DEFAULT_LOCK_AMOUNT, "Unlocked balance should match")
-    assert.equal((await staking.locksCount(owner)).valueOf(), parseInt(lockId, 10), "last lock id should match")
+    const [ staked, locked ] = await staking.getBalancesOf(owner)
+    assert.equal(staked.toString(), DEFAULT_STAKE_AMOUNT, "Staked balance should match")
+    assert.equal(locked.toString(), DEFAULT_LOCK_AMOUNT, "Locked balance should match")
   })
 
   it('fails locking 0 tokens', async () => {
     await approveAndStake()
-    await assertRevert(staking.lock(0, user1, EMPTY_STRING))
+    await assertRevert(staking.allowManagerAndLock(0, user1, 1, EMPTY_DATA))
+  })
+
+  it('fails locking without enough allowance', async () => {
+    await approveAndStake()
+    await assertRevert(staking.allowManagerAndLock(2, user1, 1, EMPTY_DATA))
```

Comment:
let's add the revert reasons

Reply:
This a problem in the whole repo, I was planning to do it all at once, see: https://github.com/aragon/staking/issues/18